src/vector_windows.rs

```diff
@@ -121,7 +167,43 @@ pub mod service_control {
         Ok(())
     }
 
-    fn install_service(service_def: &ServiceDefinition) -> Result<()> {
+    fn restart_service(service_def: &ServiceDefinition) -> crate::Result<()> {
+        let service_access =
+            ServiceAccess::QUERY_STATUS | ServiceAccess::START | ServiceAccess::STOP;
+        let service = open_service(&service_def, service_access).map_err(Error::from)?;
+        let service_status = service.query_status().map_err(Error::from)?;
+
+        if service_status.current_state == ServiceState::StartPending
+            || service_status.current_state == ServiceState::Running
+        {
+            service.stop().map_err(Error::from)?;
+        }
+
+        let timeout = Duration::from_secs(10);
+        let poll_status = poll_state(
+            &service,
+            ServiceState::Stopped,
+            timeout,
+            Duration::from_secs(1),
```

Comment:
It seems we can use a wait hint here: https://docs.rs/windows-service/0.3.0/windows_service/service/struct.ServiceStatus.html#structfield.wait_hint ?

Reply:
Two comments on this:

* Unfortunately, it looks like the `windows-service` crate does not let you specify a `wait_hint` when stopping the service. It uses a zeroe'd `SERVICE_STATUS` under the hood.
* I think I still need to poll the state of the service to determine when it is safe to start it back.

Note that I'm also using it in `uninstall` to wait for the service to be stopped before deleting it from the Service Control Manager.

Reply:
Okay! :) Very well considered reasoning, thank you.